File: frontend_flutter/fruit_market/lib/models/Payment.dart
import 'order.dart';

class Payment {
  final String paymentId;
  final String orderId;
  final double amount;
  final String? paymentMethod;
  final String paymentStatus;
  final String? transactionCode;
  final DateTime? paidAt;

  final Order? order;

  Payment({
    required this.paymentId,
    required this.orderId,
    required this.amount,
    this.paymentMethod,
    required this.paymentStatus,
    this.transactionCode,
    this.paidAt,
    this.order,
  });

  /// JSON -> Object
  factory Payment.fromJson(Map<String, dynamic> json) {
    return Payment(
      paymentId: json['paymentId'],
      orderId: json['orderId'],
      amount: (json['amount'] as num).toDouble(),
      paymentMethod: json['paymentMethod'],
      paymentStatus: json['paymentStatus'],
      transactionCode: json['transactionCode'],
      paidAt: json['paidAt'] != null ? DateTime.parse(json['paidAt']) : null,
      order: json['order'] != null ? Order.fromJson(json['order']) : null,
    );
  }

  /// Object -> JSON
  Map<String, dynamic> toJson() {
    return {
      'paymentId': paymentId,
      'orderId': orderId,
      'amount': amount,
      'paymentMethod': paymentMethod,
      'paymentStatus': paymentStatus,
      'transactionCode': transactionCode,
      'paidAt': paidAt?.toIso8601String(),
      'order': order?.toJson(),
    };
  }
}

File: backend_dotnet/fruit_api/fruit_api/Models/Payment.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace fruit_api.Models;

[Table("Payments")]
public class Payment
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    [Column("paymentId")]
    public string PaymentId { get; set; } = null!;

    [Required]
    [Column("orderId")]
    public string OrderId { get; set; } = null!;

    [Required]
    [Column("amount", TypeName = "decimal(12,2)")]
    public decimal Amount { get; set; }

    [MaxLength(50)]
    [Column("paymentMethod")]
    public string? PaymentMethod { get; set; }

    [MaxLength(30)]
    [Column("paymentStatus")]
    public string PaymentStatus { get; set; } = "unpaid";

    [MaxLength(100)]
    [Column("transactionCode")]
    public string? TransactionCode { get; set; }

    [Column("paidAt")]
    public DateTime? PaidAt { get; set; }

    // MoMo fields
    [MaxLength(100)]
    [Column("momoRequestId")]
    public string? MomoRequestId { get; set; }

    [MaxLength(50)]
    [Column("momoPartnerCode")]
    public string? MomoPartnerCode { get; set; }

    [MaxLength(500)]
    [Column("qrCodeUrl")]
    public string? QrCodeUrl { get; set; }

    // Navigation properties
    [ForeignKey("OrderId")]
    public Order? Order { get; set; }
}